### PR TITLE
feat(dependencies): update plugins and remove eslint@7 support

### PR DESCRIPTION
**Pull request by dogpatch626**
## Description

BREAKING CHANGES: 
update to eslint-plugin-unicorn 47.0.0
* [Require minimum of eslint 8 and node 16 required to fix bug in node 18](https://github.com/sindresorhus/eslint-plugin-unicorn/releases/tag/v46.0.1)

update eslint-config-prettier v8 
* [all configs merged to one](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21)

esling-plugin-prettier 
* [Drop support for eslint 5/6, prettier 1, node 6/8](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md#v400-2021-08-30)

Please make sure that the PR fulfills these requirements

- [ ] I checked that there aren't other open Pull Requests for the same update/change.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] Rule changes includes a comment to describe the reasoning behind the change.
- [ ] PR contains a single rule change.
    
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update

### CODE REVIEW

**Comment by smackfu** [comment]
@dogpatch626 I think an admin will have to update the required checks to merge this.

**Review by smackfu — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -44,23 +44,23 @@
...
-    "eslint-plugin-you-dont-need-lodash-underscore": "^6.10.0"
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-unicorn": "^47.0.0",
+    "eslint-plugin-you-dont-need-lodash-underscore": "^6.12.0"
   },
   "peerDependencies": {
-    "eslint": "^7.32.0 || ^8.8.0",
+    "eslint": "^8.42.0",
     "eslint-plugin-jest": "^24.1.3 || ^25.7.0 || ^26.0.0 || ^27.0.0",
```
[the remark this answers is not in the document]
Comment (smackfu) [reply package.json:60]: It's not clear to me why this is required based on the breaking changes in v25:
https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.0

[the remark this answers is not in the document]
Comment (smackfu) [reply package.json:60]: I'm using node v16.17.1

[the remark this answers is not in the document]
Comment (dogpatch626) [reply package.json:60]: Hey, appreciate you looking into this aswell. I was looking when I stumbled across this thread https://github.com/jest-community/eslint-plugin-jest/pull/940 and installed 25.1.0 and npm run test:lint seems to run fine so far. going to test a little more

[the remark this answers is not in the document]
Comment (smackfu) [reply package.json:60]: Oh good call, yeah, it's not a breaking change for them to support a new major of a dependency.
https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.1.0

**Review by smackfu — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Pull request merged**